pkg/server/endpoints/node/handler.go

```diff
@@ -662,6 +662,10 @@ func (h *Handler) validateAgentSVID(ctx context.Context, cert *x509.Certificate)
 		return errors.New("agent is not attested")
 	}
 
+	if n.CertSerialNumber == "" && n.NewCertSerialNumber == "" {
```

Comment:
II think we must move it into a helper, we'll be using it in different places, 
just returning a boolean